python/ml4ir/tests/test_features.py

```diff
@@ -0,0 +1,74 @@
+from ml4ir.tests.test_base import RankingTestBase
+from ml4ir.features import preprocessing
+from ml4ir.features.feature_layer import get_sequence_embedding
+import tensorflow as tf
+import string
+import numpy as np
+from ml4ir.config.keys import TFRecordTypeKey
+
+
+class RankingModelTest(RankingTestBase):
+    def test_text_preprocesing(self):
+        """
+        Unit test text preprocessing
+        """
+        preprocessing_info = {"to_lower": True, "max_length": 20, "remove_punctuation": True}
+        input_text = "ABCabc123!@#"
+        processed_bytes_tensor = preprocessing.preprocess_text(input_text, preprocessing_info)
+        processed_text = (
+            tf.strings.unicode_encode(
+                tf.cast(processed_bytes_tensor, tf.int32), output_encoding="UTF-8",
+            )
+            .numpy()
+            .decode("utf-8")
+        )
+
+        # Clipping and padding to the maximum length set
+        assert len(processed_bytes_tensor) == preprocessing_info["max_length"]
+
+        # Converting to lower case
+        assert processed_text.lower() == processed_text
+
+        # Removing punctuation
+        assert (
+            processed_text.translate(str.maketrans("", "", string.punctuation)) == processed_text
+        )
+
+        # Overall
+        assert processed_text.replace("\x00", "") == input_text.lower().translate(
+            str.maketrans("", "", string.punctuation)
+        )
+        assert processed_text.replace("\x00", "") == "abcabc123"
+
+    def test_sequence_embedding(self):
+        """
+        Unit test sequence embedding
+
+        """
+        batch_size = 50
+        max_length = 20
+        embedding_size = 128
+        feature_info = {
+            "feature_layer_info": {"embedding_size": embedding_size, "embedding_type": "bilstm"},
```

Comment:
so I'm missing a config piece here, and terminology: the "embedding" size should be the dimension of the character / token Embedding layer right after one-hot encoding (unless that is not used, but still: first step of the chain).  For us, that should be 256 (since we're not hashing, are we?).  Then there's the lstm output dimension.  The output of the bilstm will be 2 * this value.  

This value is not the "embedding" dimension, it's independent, and should be called the "text_encoding" dimension or something like that.

Reply:
Updated the implementation.